src/Build/Xml/XmlReaderExtension.cs

```diff
@@ -34,25 +34,20 @@ private XmlReaderExtension(string file)
         {
             try
             {
+                // Note: Passing in UTF8 w/o BOM into StreamReader. If the BOM is detected StreamReader will set the
+                // Encoding correctly (detectEncodingFromByteOrderMarks = true). The default is to use UTF8 (with BOM)
+                // which will cause the BOM to be added when we re-save the file in cases where it was not present on
+                // load.
                 _stream = new FileStream(file, FileMode.Open, FileAccess.Read, FileShare.Read);
-                _streamReader = new StreamReader(_stream, Encoding.UTF8, true);
+                _streamReader = new StreamReader(_stream, s_utf8NoBom, detectEncodingFromByteOrderMarks: true);
+
                 Reader = GetXmlReader(_streamReader, out _encoding);
-                var bom = _stream.StartsWithPreamble();
 
                 // Override detected encoding if xml encoding attribute is specified
                 var encodingAttribute = Reader.GetAttribute("encoding");
                 _encoding = !string.IsNullOrEmpty(encodingAttribute)
                     ? Encoding.GetEncoding(encodingAttribute)
                     : _encoding;
```

Comment:
It is set by that a bit farther down in the file and returned as an out param of GetXmlReader:
https://github.com/Microsoft/msbuild/pull/1904/files#diff-abe59969026298c02814a641a6e57fceL83